### DISCUSSION TITLE
Any way to stop post system setup from overwriting PATH?

### DISCUSSION THREAD

**Discussion by Anti-Distinctlyminty** [AcademySoftwareFoundation/rez#1491] [Q&A]
I'm taking a look at the context of a rez environment and I noticed something that is causing me some headaches regarding the `PATH` environment variable. To do this I'm running something like

```
rez-env my_package -- rez-context --interpret > my_package.bat
```

my_package make of the rez `source` command, and the file it sources makes alterations to `PATH`.  However, the result of the previous command is that I end up with a `my_package.bat` that looks like this:

```bat
call C:\ProgramData\Rez\cache\cgru\3.3.0-b75c1cb\0368\a\cgru\setup_python.cmd
REM
REM ################################################################################
REM post system setup
REM ################################################################################
...
set PATH=C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows

```

As you can see, because `PATH` is set during the post system setup any changes made by the `call` to `PATH` are overwritten.
Is there any way to get around this problem?

**Comment by Jawabiscuit** [comment]
My question would be, do you you have to do it this way in the first place? If you need to manipulate PATH to do some "setup" for python why not do the setup stuff in python's package.py, or some other python_utils package, or maybe see if ephemerals are what you need? It seems like injecting environment variables by stringing together commands is not very maintainable or understandable from a user perspective. Maybe I'm missing something. Rez comes with many features, I encourage you to look through the configuration to find something that better suits your needs.

**Reply by JeanChristopheMorinPerso** [reply]
> I encourage you to look through the configuration to find something that better suits your needs.

I would have worded that differently. I think @Anti-Distinctlyminty is asking for help. Saying "read the docs" isn't really helpful.

**Reply by Jawabiscuit** [reply]
Yeah, I'm happy to provide a better solution given more info. I agree I can leave out the RTM but reading the config actually helped me before, there's a lot that can be missed. I'd be happy to point to some configuration if know better the "why".

**Reply by Anti-Distinctlyminty** [reply]
All good. I should've been more clear that I'm implementing a package from code that is not my own that has certain constraints. Continued in my reply to @JeanChristopheMorinPerso below.

**Comment by JeanChristopheMorinPerso** [comment]
I don't think that it is supported, but your are not the first to ask for something like that. I know I have personally wanted that once or twice in the past. I've always been on the fence though and I never succeeded at convincing myself that it's a good way to manage an environment.

@Anti-Distinctlyminty can you tell us more about what you are trying to achieve? We can probably help you find an alternative solution and it would also help us collect more data on how our users use rez (which generally helps to build the roadmap and plan features).

**Reply by Anti-Distinctlyminty** [reply]
I am creating a package for [CGRU](https://cgru.info/) and as part of its setup you must call a cmd file (which in turn call other cmd files). It's just the way it is designed, so there's just not much room for alternatives there. Using the `source` command works for every environment variable with the exception of `PATH` in my case, however, I also tested that any use of `env.append` or `env.prepend` will cause the same to happen to any environment variable, not just `PATH`.

The only alternative I can think of is to replicate the function cmd files in `commands()` and not source the cmd files at all but that would be quite laborious.

I don't suppose there's any way I can perform a post-post-system-setup?
